src/zhinst/toolkit/control/node_tree.py

```diff
@@ -515,14 +519,22 @@ def dictify(data, keys: List, val: Dict) -> Dict:
     Helper function to generate nested dictionary from list of keys and value.
     Calls itself recursively.
 
+    An underscore will be appended to keys that are identical to reserved keywords
+    in python. (e.g. in -> in_)
+
     Arguments:
         data (dict): A dictionary to add value to with keys.
         keys (list): A list of keys to traverse along tree and place value.
         val (dict): A value for innermost layer of nested dict.
 
     """
     key = keys[0]
-    key = int(key) if key.isdecimal() else key.lower()
+    if key.isdecimal():
+        key = int(key)
+    else:
+        key = key.lower()
+        if keyword.iskeyword(key):
+            key = key + "_"
```

Comment:
Nit: I find it easier to understand the logic like this:
```suggestion
        key += "_" if keyword.iskeyword(key) else ""
```